src/app/GitCommands/RevisionReader.cs

```diff
@@ -184,13 +201,21 @@ public IReadOnlyCollection<GitRevision> GetRevisionsFromRange(string olderCommit
                 Debug.WriteLine($"git {arguments}");
 #endif
                 using IProcess process = _module.GitCommandRunner.RunDetached(cancellationToken, arguments, redirectOutput: true, outputEncoding: null);
-                string errorOutput = process.StandardError.ReadToEnd();
+
+                // StandardError.ReadToEnd[Async] hangs if lengthy StandardOutput is not consumed in parallel. So, buffer the StandardOutput
+                MemoryStream standardOutputBuffer = new();
+                Task standardOutputTask = process.StandardOutput.BaseStream.CopyToAsync(standardOutputBuffer, cancellationToken);
+                Task<string> errorOutputTask = process.StandardError.ReadToEndAsync(cancellationToken);
+
+                string errorOutput = await errorOutputTask;
                 if (!string.IsNullOrWhiteSpace(errorOutput) && throwOnError)
                 {
                     throw new ExternalOperationException(AppSettings.GitCommand, arguments.ToString(), innerException: new Exception(errorOutput));
                 }
 
-                commandBytes = new ReadOnlyMemory<byte>(process.StandardOutput.BaseStream.SplitLogOutput().SingleOrDefault().ToArray());
+                await standardOutputTask;
+                standardOutputBuffer.Position = 0;
+                commandBytes = new ReadOnlyMemory<byte>(standardOutputBuffer.SplitLogOutput().SingleOrDefault().ToArray());
```

Comment:
Please annotate (at least method) with NRT (you can use `#nullable enable/disable` in code).

Does `new ReadOnlyMemory<byte>()` accept null?

Reply:
> Please annotate (at least method) with NRT (you can use `#nullable enable/disable` in code).

Done in a separate commit.

> Does `new ReadOnlyMemory<byte>()` accept null?

I does not need to. What could be `null` here? Proven by `﻿#nullable enable`.

Reply:
SingleOrDefault suggests it may be a null...

Reply:
You are right that this code was suspicious and that `default` often means `null`.
But the `default` for `ReadOnlyMemory<byte>` seems to be an instance with length 0.
And its ctor accepts `byte[]?`.
But this was completely unnecessary because `SplitLogOutput` already yields the needed type.